src/Symfony/Component/Uid/Factory/MockUuidFactory.php

```diff
@@ -0,0 +1,143 @@
+use Symfony\Component\Uid\Exception\LogicException;
+use Symfony\Component\Uid\TimeBasedUidInterface;
+use Symfony\Component\Uid\Uuid;
+use Symfony\Component\Uid\UuidV1;
+use Symfony\Component\Uid\UuidV3;
+use Symfony\Component\Uid\UuidV4;
+use Symfony\Component\Uid\UuidV5;
+use Symfony\Component\Uid\UuidV6;
+use Symfony\Component\Uid\UuidV7;
+
+class MockUuidFactory extends UuidFactory
+{
+    private \Iterator $sequence;
+
+    /**
+     * @param iterable<string|Uuid> $uuids
+     */
+    public function __construct(
+        iterable $uuids,
+        private Uuid|string|null $timeBasedNode = null,
+        private Uuid|string|null $nameBasedNamespace = null,
+    ) {
+        $this->sequence = match (true) {
+            \is_array($uuids) => new \ArrayIterator($uuids),
+            $uuids instanceof \Iterator => $uuids,
+            $uuids instanceof \Traversable => new \IteratorIterator($uuids),
+        };
+    }
+
+    public function create(): Uuid
+    {
+        if (!$this->sequence->valid()) {
+            throw new LogicException('No more UUIDs in sequence.');
+        }
+        $uuid = $this->sequence->current();
+        $this->sequence->next();
+
+        return match (true) {
+            $uuid instanceof Uuid => $uuid,
+            \is_string($uuid) => Uuid::fromString($uuid),
+            default => throw new InvalidArgumentException(\sprintf('Next UUID in sequence is not a valid UUID string or object: "%s" given.', get_debug_type($uuid))),
+        };
+    }
+
+    public function randomBased(): RandomBasedUuidFactory
+    {
+        return new class($this->create(...)) extends RandomBasedUuidFactory {
+            public function __construct(
+                private \Closure $create,
+            ) {
+            }
+
+            public function create(): UuidV4
+            {
+                if (!($uuid = ($this->create)()) instanceof UuidV4) {
+                    throw new InvalidArgumentException(\sprintf('Next UUID in sequence is not a UuidV4: "%s" given.', get_debug_type($uuid)));
+                }
+
+                return $uuid;
+            }
+        };
+    }
+
+    public function timeBased(Uuid|string|null $node = null): TimeBasedUuidFactory
+    {
+        if (\is_string($node ??= $this->timeBasedNode)) {
+            $node = Uuid::fromString($node);
+        }
+
+        return new class($this->create(...), $node) extends TimeBasedUuidFactory {
+            public function __construct(
+                private \Closure $create,
+                private ?Uuid $node = null,
+            ) {
+            }
+
+            public function create(?\DateTimeInterface $time = null): Uuid&TimeBasedUidInterface
+            {
+                if (!($uuid = ($this->create)()) instanceof TimeBasedUidInterface) {
+                    throw new InvalidArgumentException(\sprintf('Next UUID in sequence is not a TimeBasedUidInterface: "%s" given.', get_debug_type($uuid)));
```

Comment:
to help static analyzers:
```suggestion
                if (!($uuid = ($this->create)()) instanceof TimeBasedUidInterface || !$uuid instanceof Uuid) {
                    throw new InvalidArgumentException(\sprintf('Next UUID in sequence is not a TimeBasedUidInterface nor a Uuid: "%s" given.', get_debug_type($uuid)));
```